### PR TITLE
Support for new collation options for GPDB7+

### PR DESCRIPTION
**Pull request by shivzone**
In GPDB 7+, users can now specify whether the collation should use deterministic comparisons when creating a collation and a provider to use for locale services. This commit supports these 2 new options to be backed up.

Pending: Tests

### CODE REVIEW

File: backup/queries_types.go [hunk 1]
```diff
@@ -554,15 +556,32 @@ func (c Collation) FQN() string {
...
 		JOIN pg_namespace n ON c.collnamespace = n.oid
 	WHERE %s`, SchemaFilterClause("n"))
+	} else {
+		query = fmt.Sprintf(`
+	SELECT c.oid,
+		quote_ident(n.nspname) AS schema,
+		quote_ident(c.collname) AS name,
+		c.collcollate AS collate,
+		c.collctype AS ctype,
+		c.collprovider as provider,
```
Comment (roicos) [inline backup/queries_types.go:579]: `c.collprovider as provider,
  c.collisdeterministic as IsDeterministic`

Those fields are not present in GPDB 6. Provider was introduced in PG10 and deterministic- in PG12.

File: backup/queries_types.go [hunk 2]
```diff
@@ -554,15 +556,32 @@ func (c Collation) FQN() string {
 }
 
 func GetCollations(connectionPool *dbconn.DBConn) []Collation {
```
Comment (roicos) [inline backup/queries_types.go:558]: Any tests changes?

File: backup/predata_types.go [hunk 3]
```diff
@@ -205,7 +207,26 @@ func PrintCreateRangeTypeStatement(metadataFile *utils.FileWithByteCount, toc *t
 func PrintCreateCollationStatements(metadataFile *utils.FileWithByteCount, toc *toc.TOC, collations []Collation, collationMetadata MetadataMap) {
```
Comment (roicos) [inline backup/predata_types.go:207]: Any tests changes for this function?

File: backup/queries_types.go (hunk 2 above)
Comment (shivzone) [reply backup/queries_types.go:558]: tests changes are pending in this PR

File: backup/predata_types.go (hunk 3 above)
Comment (shivzone) [reply backup/predata_types.go:207]: tests changes are pending in this PR

File: backup/queries_types.go (hunk 1 above)
Comment (shivzone) [reply backup/queries_types.go:579]: my bad i wasn't supposed to include the 2 new fields in this block

File: backup/queries_types.go [hunk 4]
```diff
@@ -554,15 +556,32 @@ func (c Collation) FQN() string {
...
-	query := fmt.Sprintf(`
+	var query string
+	if connectionPool.Version.AtLeast("7") {
```
[the remark this answers is not in the document]
Comment (shivzone) [reply backup/queries_types.go:560]: this has been addressed. The query for GPDB <=6 has been modified

**Comment by shivzone** [comment]
tests have been updated to cover new collation options

**Review by roicos — APPROVED** [review]

**Pull request merged**